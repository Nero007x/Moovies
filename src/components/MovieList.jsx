import React, { useState } from 'react';
import { View, FlatList, Image, Text, StyleSheet, TouchableOpacity, ActivityIndicator } from 'react-native';
import { useMoviesQuery } from '../service/useMoviesQuery'; // Adjust the path as necessary

const MovieList = ({ navigation }) => {
  const [page, setPage] = useState(1); // State for pagination

  // Correctly destructure the response from the query hook
  const { data: movies, isLoading, error, refetch } = useMoviesQuery(page);

  const renderItem = ({ item }) => (
    <TouchableOpacity onPress={() => navigation.navigate('MovieDetails', { movie: item })}>
      <View style={styles.movieItem}>
        <Image
          source={{ uri: `https://image.tmdb.org/t/p/w500${item.poster_path}` }}
          style={styles.poster}
        />
        <Text style={styles.title}>{item.title}</Text>
      </View>
    </TouchableOpacity>
  );

  // Show loading indicator while fetching
  if (isLoading) {
    return <ActivityIndicator size="large" color="#0000ff" />;
  }

  // Handle error state using the updated "error" variable
  if (error) {
    return (
      <View>
        <Text>Error fetching movies.</Text>
        <TouchableOpacity onPress={refetch}>
          <Text>Retry</Text>
        </TouchableOpacity>
      </View>
    );
  }

  // Render the movie list
  return (
    <FlatList
      data={movies}
      renderItem={renderItem}
      keyExtractor={(item) => item.id.toString()}
      onEndReached={() => setPage((prevPage) => prevPage + 1)} // Load more data when reaching the end
      onEndReachedThreshold={0.5}
    />
  );
};

const styles = StyleSheet.create({
  movieItem: {
    flexDirection: 'row',
    margin: 10,
  },
  poster: {
    width: 100,
    height: 150,
  },
  title: {
    marginLeft: 10,
    fontSize: 18,
    alignSelf: 'center',
  },
});

export default MovieList;