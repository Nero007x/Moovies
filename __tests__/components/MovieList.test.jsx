import React from 'react';
import { render, screen } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import MovieList from '../../src/components/MovieList';

const mockMovies = [
  {
    id: 1,
    title: 'Inception',
    poster_path: '/poster_path_inception.jpg',
  },
  {
    id: 2,
    title: 'Interstellar',
    poster_path: '/poster_path_interstellar.jpg',
  },
];

const mockNavigation = {
  navigate: jest.fn(),
};

test('MovieList component renders correctly', async () => {
  const mock = new MockAdapter(axios);
  mock.onGet(/now_playing/).reply(200, { results: mockMovies });

  let component;
  await renderer.act(async () => {
    component = renderer.create(<MovieList navigation={mockNavigation} />);
  });

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

