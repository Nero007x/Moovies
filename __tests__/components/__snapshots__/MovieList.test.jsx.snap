// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component renders correctly 1`] = `
<RCTScrollView
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;
